Dockerfile

```diff
@@ -1,19 +1,57 @@
-FROM python:3.12.10-alpine3.21@sha256:9c51ecce261773a684c8345b2d4673700055c513b4d54bc0719337d3e4ee552e
+FROM python:3.12.10-alpine3.21@sha256:9c51ecce261773a684c8345b2d4673700055c513b4d54bc0719337d3e4ee552e AS base
 
-WORKDIR /ldap-jwt-auth-run
+# Install python-ldap system dependencies
+RUN apk add --no-cache build-base openldap-dev
+
+WORKDIR /app
 
 COPY pyproject.toml requirements.txt ./
 COPY ldap_jwt_auth/ ldap_jwt_auth/
 
+
+FROM base AS dev
+
+WORKDIR /app
+
 RUN --mount=type=cache,target=/root/.cache \
     set -eux; \
     \
-    # Install python-ldap system dependencies \
-    apk add --no-cache build-base openldap-dev; \
-    \
-    python -m pip install .[dev]; \
+    pip install --no-cache-dir .[dev]; \
     # Ensure the pinned versions of the production dependencies and subdependencies are installed \
-    python -m pip install --no-cache-dir --requirement requirements.txt;
+    pip install --no-cache-dir --requirement requirements.txt;
 
 CMD ["fastapi", "dev", "ldap_jwt_auth/main.py", "--host", "0.0.0.0", "--port", "8000"]
+
+EXPOSE 8000
+
+
+FROM dev AS test
+
+WORKDIR /app
+
+COPY test/ test/
+
+CMD ["pytest",  "--config-file", "test/pytest.ini", "-v"]
+
+
+FROM base AS prod
+
+WORKDIR /app
+
+RUN --mount=type=cache,target=/root/.cache \
+    set -eux; \
+    \
+    # Ensure the package gets installed properly using the pyproject.toml file \
+    pip install --no-cache-dir .; \
```

Comment:
We do this in IMS API otherwise the `ims-migrate` script does not get installed.